internal/services/users-service.go

```diff
@@ -56,6 +57,22 @@ func (s *UsersService) NewUser(input ports.CreateUserInput) *customErrors.Intern
 	return nil
 }
 
+func (s *UsersService) GetUser(id string) (*entities.User, *customErrors.InternalError) {
+	if !validators.IsValidUuid(id) {
+		return nil, customErrors.NewInternalError("invalid uuid", 400, []string{})
+	}
+
+	response, err := s.repository.FindById(id)
+
+	if err != nil {
+		return nil, customErrors.NewInternalError("a failure occurred when try to retrieve a new user", 500, []string{err.Error()})
```

Comment:
Apesar de existir a opção de detalhes no erro, não é uma boa prática detalhar o motivo de um erro interno para fora. Se precisarmos descobrir a causa, é melhor ir com o debug nesse ponto e ver a mensagem.
```suggestion
		return nil, customErrors.NewInternalError("a failure occurred when try to retrieve a new user", 500, []string{})
```